ecs-init/cache/cache_test.go

```diff
@@ -99,6 +102,46 @@ func TestIsAgentCachedTrue(t *testing.T) {
 	}
 }
 
+func TestAgentCacheStatus(t *testing.T) {
+	var cases = []struct {
+		data     string
+		expected CacheStatus
+	}{
+		// Expected states:
+		{"0", StatusUncached},
+		{"1", StatusCached},
+		{"2", StatusReloadNeeded},
+		{"1\n", StatusCached},
+		// Invalid states:
+		{"spurious", StatusUncached},
+		{" ", StatusUncached},
+		{"256", StatusUncached},
+	}
+
+	for _, testcase := range cases {
+		t.Run(string(testcase.data), func(t *testing.T) {
+			mockCtrl := gomock.NewController(t)
+			defer mockCtrl.Finish()
+
+			file := ioutil.NopCloser(bytes.NewBufferString(testcase.data))
+			mockFS := NewMockfileSystem(mockCtrl)
+			mockFSInfo := NewMockfileSizeInfo(mockCtrl)
+
+			mockFS.EXPECT().Stat(config.CacheState()).Return(mockFSInfo, nil)
+			mockFS.EXPECT().Stat(config.AgentTarball()).Return(mockFSInfo, nil)
+			mockFSInfo.EXPECT().Size().Return(int64(1)).Times(2)
+			mockFS.EXPECT().Open(config.CacheState()).Return(file, nil)
+
+			d := &Downloader{fs: mockFS}
+
+			actual := d.AgentCacheStatus()
+			if actual != testcase.expected {
```

Comment:
nit: you can just do `assert.Equals` here (and in other tests)?

Reply:
I could, however I was avoiding adding new dependencies.